tests/frontend/callbacks/test_variables_callbacks.py

```diff
@@ -310,76 +310,131 @@ def test_populate_variables_workspace_filter_variables(
+        ),
+        (
+            "01",
+            DatasetIdentifiers.DATA_SOURCE,
+            VariableIdentifiers.DATA_SOURCE,
+        ),
+        (
+            "2009-01-02",
+            DatasetIdentifiers.CONTAINS_DATA_FROM,
+            VariableIdentifiers.CONTAINS_DATA_FROM,
+        ),
+        (
+            "2021-08-10",
+            DatasetIdentifiers.CONTAINS_DATA_UNTIL,
+            VariableIdentifiers.CONTAINS_DATA_UNTIL,
+        ),
+    ],
+)
+def test_update_variables_values_from_dataset_values(
+    dataset_value,
+    dataset_identifier,
+    variable_identifier,
+    metadata: DataDocMetadata,
+):
     state.metadata = metadata
-    dataset_temporality_type = "FIXED"
-    dataset_data_source = None
-    setattr(
-        state.metadata.dataset,
-        DatasetIdentifiers.TEMPORALITY_TYPE,
-        dataset_temporality_type,
-    )
     setattr(
         state.metadata.dataset,
-        DatasetIdentifiers.DATA_SOURCE,
-        dataset_data_source,
+        dataset_identifier,
+        dataset_value,
     )
     set_variables_values_inherited_from_dataset(
-        dataset_temporality_type,
-        DatasetIdentifiers.TEMPORALITY_TYPE,
+        dataset_value,
+        dataset_identifier,
     )
     for val in state.metadata.variables:
-        assert metadata.dataset.temporality_type == get_standard_metadata(
+        assert dataset_value == get_metadata_and_stringify(
             metadata.variables_lookup[val.short_name],
-            VariableIdentifiers.TEMPORALITY_TYPE.value,
-        )
-    set_variables_values_inherited_from_dataset(
-        dataset_data_source,
-        DatasetIdentifiers.DATA_SOURCE,
-    )
-    for val in state.metadata.variables:
-        assert metadata.dataset.data_source == get_standard_metadata(
-            metadata.variables_lookup[val.short_name],
-            VariableIdentifiers.DATA_SOURCE.value,
+            variable_identifier.value,
         )
 
 
-def test_variables_value_can_be_changed_after_update_from_dataset_value(
+@pytest.mark.parametrize(
+    (
+        "dataset_value",
+        "dataset_identifier",
+        "variable_identifier",
+        "update_value",
+    ),
+    [
+        (
+            "EVENT",
+            DatasetIdentifiers.TEMPORALITY_TYPE,
+            VariableIdentifiers.TEMPORALITY_TYPE,
+            "ACCUMULATED",
+        ),
+        (
+            "02",
+            DatasetIdentifiers.DATA_SOURCE,
+            VariableIdentifiers.DATA_SOURCE,
+            "03",
+        ),
+        (
+            "2009-01-02",
+            DatasetIdentifiers.CONTAINS_DATA_FROM,
+            VariableIdentifiers.CONTAINS_DATA_FROM,
+            "1998-03-11",
+        ),
+        (
+            "1988-11-03",
+            DatasetIdentifiers.CONTAINS_DATA_UNTIL,
+            VariableIdentifiers.CONTAINS_DATA_UNTIL,
+            "2008-01-01",
+        ),
+    ],
+)
+def test_variables_values_can_be_changed_after_update_from_dataset_value(
```

Comment:
I'm not quite sure what this is testing, is there a risk that the variables values couldn't be updated?

Reply:
It was a risk when developing and important to check that the inherited value was persisted in rest of the variables even after individual values were changed. But whether or not it is still important to test for this now that the code works I don't know...